assets/service_now_api_gateway_authorizer/index.py

```diff
@@ -0,0 +1,62 @@
+import json
+import boto3
+import os
+import logging
+
+logger = logging.getLogger()
+logger.setLevel(logging.INFO)
+
+secrets_client = boto3.client('secretsmanager')
+
+def handler(event, context):
+    """
+    Lambda authorizer for ServiceNow API Gateway
+    """
+    try:
+        # Get the authorization token from the event
+        token = event.get('authorizationToken', '')
+        
+        # Remove 'Bearer ' prefix if present
+        if token.startswith('Bearer '):
+            token = token[7:]
+        
+        # Get expected token from Secrets Manager
+        api_auth_secret_arn = os.environ.get('API_AUTH_SECRET')
+        if not api_auth_secret_arn:
+            logger.error("API_AUTH_SECRET environment variable not set")
+            raise Exception('Unauthorized')
+        
+        try:
+            response = secrets_client.get_secret_value(SecretId=api_auth_secret_arn)
+            secret_dict = json.loads(response['SecretString'])
+            expected_token = secret_dict.get('token')
+        except Exception as e:
+            logger.error(f"Failed to retrieve secret: {str(e)}")
+            raise Exception('Unauthorized')
```

Comment:
suggestion(non-blocking): A more graceful failure approach is to return a policy like the below

```
  {
            'principalId': 'service-now',
            'policyDocument': {
                'Version': '2012-10-17',
                'Statement': [
                    {
                        'Action': 'execute-api:Invoke',
                        'Effect': 'Deny',
                        'Resource': event['methodArn']
                    }
                ]
          }
      }
```